src/main/java/com/azure/cosmos/kafka/connect/sink/BulkWriter.java

```diff
@@ -147,4 +168,33 @@ protected BulkOperationFailedException(int statusCode, int subStatusCode, String
             BridgeInternal.setSubStatusCode(this, subStatusCode);
         }
     }
+
+    private static class IdAndPartionKey {
+        String id;
+        PartitionKey partitionKey;
+
+        public IdAndPartionKey(String id, PartitionKey partitionKey) {
+            this.id = id;
+            this.partitionKey = partitionKey;
+        }
+
+
+        public String getId() {
+            return id;
+        }
+
+        public PartitionKey getPartitionKey() {
+            return partitionKey;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
```

Comment:
In order for all equality operations to work properly, we will also need to implement hashCode() method for this class. You can read more about the contract here - https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#equals-java.lang.Object-